mix/CodeModel.cpp

```diff
@@ -93,8 +93,24 @@ class CollectDeclarationsVisitor: public ASTConstVisitor
 	bool m_functionScope;
 	uint m_storageSlot;
 };
+
+dev::eth::AssemblyItems filterLocations(dev::eth::AssemblyItems const& _locations, dev::solidity::ContractDefinition const& _contract, QHash<LocationPair, QString> _functions)
+{
+	dev::eth::AssemblyItems result;
+	result.reserve(_locations.size());
+	std::string sourceName = *_contract.getLocation().sourceName;
+	for (dev::eth::AssemblyItem item : _locations)
+	{
+		dev::SourceLocation const& l = item.getLocation();
+		if (sourceName != *l.sourceName || _contract.getLocation() == l || _functions.contains(LocationPair(l.start, l.end)))
```

Comment:
We are also filtering using the sourceName? because at some point it could be interesting to highlight the code in the targeted sourceName (if available).